Assignment_2-data-emma-luk.js

```diff
@@ -0,0 +1,101 @@
+﻿// Emma Luk
+// Assignments 2 Using the revealing module pattern, build a Date utility module 
+
+var myDate = (function iife() {
+    var displaysModuleDate = null;
+
+    //private function: date instantiation
+    function _initDate() {
+        return new Date();
+    }
+
+    //check here that date is set: if not, initialize date to now
+    function setDate(date) {
+        if (date === undefined && null) {
+            displaysModuleDate = _initDate();
+        }
+        else if (typeof date === 'number' && !isNaN(date)) {
+            displaysModuleDate = new Date(date);
+        }
+        else if (typeof date === 'object') {
+            displaysModuleDate = date;
+        }
+        else {
+            console.log("sorry!! wrong type");
+        }
+    }
+
+    function getDate(objFormat) {
+        if (displaysModuleDate === null && undefined) {
+            displaysModuleDate = _initDate();
+        }
+        if (objFormat.format === "formatted") {
+            return displaysModuleDate.getMonth() + " " + displaysModuleDate.getDate() + ", " + displaysModuleDate.getFullYear();
+        }
+        else if (objFormat.format === "milliseconds" || objFormat === undefined) {
+            return displaysModuleDate.getDate();
+        }
+        else {
+            console.log("sorry!! unknown format");
+        }
+    }
+
+    function getDayName() {
+        var day = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+        if (displaysModuleDate === null) {
+            return day[_initDate().getDay()];
+        }
+
+        return day[displaysModuleDate.getDay()];
+    }
+
+    function getMonthName() {
+        var month = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

```

Comment:
Same comment here as above with respect to the getDayName method. Put your array outside the method so it doesn't re-declare and set aside memory for it each time the method is called.